drivers/dma/dma_stm32.c

```diff
@@ -257,14 +257,16 @@ static int dma_stm32_configure(struct device *dev, u32_t id,
 		return -EINVAL;
 	}
 
-	if ((stream->direction == MEMORY_TO_MEMORY) &&
+#ifdef CONFIG_DMA_STM32_V1
+if ((stream->direction == MEMORY_TO_MEMORY) &&
 		(!dev_config->support_m2m)) {
 		LOG_ERR("Memcopy not supported for device %s",
 			dev->config->name);
 		return -ENOTSUP;
 	}
-
- #ifdef CONFIG_SOC_SERIES_STM32WBX
+#endif /* CONFIG_DMA_STM32_V1 */
+	
+#ifdef CONFIG_DMA_STM32_V2
```

Comment:
This restricts V2C to mem2 mem operation, which is not what is advertised in commit message.

Reply:
for DMA_V2 this part is not needed as the stream direction is not set yet
and the `dev_config->support_m2m` is not supposed to be set and is just ignored